### PR TITLE
MM-31061: Remove pointers to slice

### PR DESCRIPTION
**Pull request by agnivade**
This PR removes instances of pointers
to slices in the codebase. There are some other instances in app/import_functions.go
but that's necessary to prevent empty arrays from appearing in the JSON output.

```release-note
NONE
```

https://mattermost.atlassian.net/browse/MM-31061

### CODE REVIEW

**Review by nronas — APPROVED** [review]
LGTM 🚀

File: services/filesstore/mocks/FileBackend.go [hunk 1]
```diff
@@ -73,16 +73,37 @@ func (_m *FileBackend) FileExists(path string) (bool, error) {
 	return r0, r1
 }
 
+// FileSize provides a mock function with given fields: path
+func (_m *FileBackend) FileSize(path string) (int64, error) {
```
Comment (isacikgoz) [inline services/filesstore/mocks/FileBackend.go:77]: Is this related?

**Review by isacikgoz — COMMENTED** [review]
Awesome 👍 Just a thing below:

File: services/filesstore/mocks/FileBackend.go (hunk 1 above)
Comment (agnivade) [reply services/filesstore/mocks/FileBackend.go:77]: This got generated from `make filestore-mocks`. Since we don't validate this in CI, this gets outdated from time to time.

**Review by isacikgoz — APPROVED** [review]
